packages/atomic/cypress/utils/analyticsUtils.ts

```diff
@@ -0,0 +1,87 @@
+import {CoveoAnalyticsClient} from '@coveo/headless/node_modules/coveo.analytics';
+import {findLast} from './arrayUtils';
+
+export type SearchEventRequest = Parameters<
+  CoveoAnalyticsClient['sendSearchEvent']
+>[0];
+
+export type ClickEventRequest = Parameters<
+  CoveoAnalyticsClient['sendClickEvent']
+>[0];
+
+export type CustomEventRequest = Parameters<
+  CoveoAnalyticsClient['sendCustomEvent']
+>[0];
+
+export type AnyEventRequest =
+  | SearchEventRequest
+  | ClickEventRequest
+  | CustomEventRequest;
+
+function isSearchEventRequest(
+  request: AnyEventRequest
+): request is SearchEventRequest {
+  return 'actionCause' in request;
+}
+
+function isClickEventRequest(
+  request: AnyEventRequest
+): request is ClickEventRequest {
+  return 'documentUri' in request;
+}
+
+function isCustomEventRequest(
+  request: AnyEventRequest
+): request is CustomEventRequest {
+  return 'eventType' in request;
+}
+
+const analyticsEventsKey = '_analyticsEvents';
+
+interface WindowWithAnalytics extends Window {
+  [analyticsEventsKey]?: AnyEventRequest[];
+}
+
+export class AnalyticsTracker {
```

Comment:
It would be ideal if we can continue using Cypress' alias features. If the root cause is the test being too fast for the UI, I'm hoping the check-point approach helps 🤞.

Reply:
The checkpoint may help. but I still see a lot of potential for race conditions whenever we wait both for a search and for an analytics call.

IMO, I would prefer to keep that utility. I don't like that some assertions have to wait for a analytics call, and I feel like using this utility would be more straightforward.